src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/resource/ReactivateAlertCodeIntTest.kt

```diff
@@ -0,0 +1,203 @@
+      .headers(setAlertRequestContext())
+      .exchange()
+      .expectStatus().isForbidden
+  }
+
+  @Test
+  fun `400 bad request - username not supplied`() {
+    val response = webTestClient.patch()
+      .uri("/alert-codes/VI/reactivate")
+      .headers(setAuthorisation(roles = listOf(ROLE_ALERTS_ADMIN)))
+      .exchange()
+      .expectStatus().isBadRequest
+      .expectBody(ErrorResponse::class.java)
+      .returnResult().responseBody
+
+    with(response!!) {
+      assertThat(status).isEqualTo(400)
+      assertThat(errorCode).isNull()
+      assertThat(userMessage)
+        .isEqualTo("Validation failure: Could not find non empty username from user_name or username token claims or Username header")
+      assertThat(developerMessage)
+        .isEqualTo("Could not find non empty username from user_name or username token claims or Username header")
+      assertThat(moreInfo).isNull()
+    }
+  }
+
+  @Test
+  fun `400 bad request - username not found`() {
+    val response = webTestClient.patch()
+      .uri("/alert-codes/VI/reactivate")
+      .headers(setAuthorisation(roles = listOf(ROLE_ALERTS_ADMIN)))
+      .headers(setAlertRequestContext(username = USER_NOT_FOUND))
+      .exchange()
+      .expectStatus().isBadRequest
+      .expectBody(ErrorResponse::class.java)
+      .returnResult().responseBody
+
+    with(response!!) {
+      assertThat(status).isEqualTo(400)
+      assertThat(errorCode).isNull()
+      assertThat(userMessage).isEqualTo("Validation failure: User details for supplied username not found")
+      assertThat(developerMessage).isEqualTo("User details for supplied username not found")
+      assertThat(moreInfo).isNull()
+    }
+  }
+
+  @Test
+  fun `404 alert type not found`() {
+    val response = webTestClient.patch()
+      .uri("/alert-codes/ALK/reactivate")
+      .headers(setAuthorisation(roles = listOf(ROLE_ALERTS_ADMIN)))
+      .headers(setAlertRequestContext())
+      .exchange()
+      .expectStatus().isNotFound
+      .expectBody(ErrorResponse::class.java)
+      .returnResult().responseBody
+
+    with(response!!) {
+      assertThat(status).isEqualTo(404)
+      assertThat(errorCode).isNull()
+      assertThat(userMessage).isEqualTo("Not found: Alert with code ALK could not be found")
+      assertThat(developerMessage)
+        .isEqualTo("Alert with code ALK could not be found")
+      assertThat(moreInfo).isNull()
+    }
+  }
+
+  @Test
+  fun `should mark alert type as active`() {
+    val alertCode = createAlertCode()
+    alertCodeRepository.findByCode(alertCode.code)!!.apply {
+      deactivatedAt = LocalDateTime.of(2010, 3, 7, 16, 27, 58)
+      deactivatedBy = "MODIFIED_BY"
+      alertCodeRepository.save(this)
+    }
+
+    val response = webTestClient.reactivateAlertCode(alertCode = alertCode.code)
+    assertThat(response.isActive).isEqualTo(true)
+
+    alertCodeRepository.findByCode(alertCode.code)!!.apply {
```

Comment:
`with` is probably more appropriate here rather than `apply` as apply is used when you're changing something about the object. Really minor though